src/storage/query/QueryBaseProcessor.h

```diff
@@ -34,6 +37,32 @@ struct Bucket {
 
 using OneVertexResp = std::tuple<PartitionID, VertexID, kvstore::ResultCode>;
 
+class ReservoirSampling final {
+public:
+    ReservoirSampling() {
+        rng_.seed(nebula::time::WallClock::fastNowInMicroSec());
+    }
+
+    int64_t sampling() {
+        ++cnt_;
+        if (cnt_ < FLAGS_max_edge_returned_per_vertex) {
+            return cnt_;
+        } else {
+            boost::random::uniform_int_distribution<> dist(0, cnt_);
+            int64_t index = dist(rng_);
+            if (index < FLAGS_max_edge_returned_per_vertex) {
+                return index;
+            } else {
+                return -1;
+            }
+        }
+    }
+
+private:
+    boost::random::mt19937  rng_;
```

Comment:
Why not use PRNG from the STL?
Why not use a 64bit generator?
IMO. `dist` should be reused.

Reply:
The distribution changes as the sample space grows, it can not be reused.